src/exposurelog/routers/edit_message.py

```diff
@@ -2,21 +2,45 @@
 
 __all__ = ["edit_message"]
 
-import typing
-
-import aiohttp
 import astropy.time
-import graphql
+import fastapi
 import sqlalchemy as sa
 
-from exposurelog.dict_from_result_proxy import dict_from_result_proxy
+from ..message import ExposureFlag, Message
+from ..shared_state import SharedState, get_shared_state
+
+router = fastapi.APIRouter()
 
 
+@router.get("/edit_message/", response_model=Message)
```

Comment:
PATCH or POST here

Reply:
Thank you for a very helpful review.

Regarding the unit tests: I went with unittest because asyncio support is built in, whereas the pytest support is not (and requires a 3rd party library that is not in release status). I'm not happy about it, but I'm even less happy with pytest's lack of built-in support for asyncio.

I have implemented your other requested changes (using POST everywhere except find_messages).

I would very much appreciate it if you could take another close look at the Dockerfile as it had some differences from gafaelfawr that I am not sure about.

Reply:
Also the github test now fails with the new Dockerfile. Based on the error message I suspect the base image you suggested does not have an adequate C++ compiler.

Reply:
For what it's worth, pytest-asyncio, despite not having a 1.x version number for some reason, is a dependency of over 10,000 other GitHub repositories and over 800 released Python packages on GitHub, so it's probably more reliable as a dependency than a lot of things with higher versions. I've only had one minor problem with it that was fixed within a week and only required avoiding one buggy release, and it's fairly actively maintained.

But unittest certainly works, and I too wish pytest would just add native async support.

Reply:
Add `build-essential` to the packages to install in `scripts/install-base-packages.sh` to fix that problem.

Reply:
(Or if you want to be really fancy, take a look at Gafaelfawr and see how I split apt package installation into two phases, one for the base image and one for the build image, and install `build-essential` via `install-dependency-packages.sh`.)